src/Symfony/Component/DependencyInjection/Alias.php

```diff
@@ -92,7 +92,7 @@ public function isPrivate()
      *
      * @throws InvalidArgumentException when the message template is invalid
      */
-    public function setDeprecated(bool $status = true, string $template = null)
+    public function setDeprecated(bool $status = true, string $template = null, string $package = null, string $sinceVersion = null)
```

Comment:
The way this is implemented is a BC break, but be can find a way around.
What about a new method: `setDeprecation(string $package, string $version, string $message, string $id, ...$args): $this`? I'm not sure we need a way to unset a deprecation here so no need to find a way to preserve this capability IMHO.

Reply:
triggerDeprecation
addDeprecation 
raiseDeprecation 

Sounds more like an action which is taken instead of just “set” sth.

Just an idea

Reply:
@OskarStark I think `set` make more sense here, as it just store the information about the deprecation not raising it